pages/docs/concepts/asyncapi-document/resuable-parts.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Reusable parts
+weight: 249
+---
+
+Reusable parts in AsyncAPI provide flexibility, modularity, and code reusability. Using reusable parts makes it easier to generate code and validate the specifications. You can reuse specific document sections such as Messages or schema definitions.
+
+Reusable parts allow you to split up the AsyncAPI document into many files and reference them using the Reference Object ($ref). You can use the $ref keyword to reference the same or another local file or external URL.
+
+## Same file
+
+You can use the ref keyword to reference a component within the same document. The image below defines referencing a component within the same document.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[MyMessageSchema]
+B[components]
+C[myChannel]
+D[$ref keyword]
+
+C -->|publishes| D
+D -->|references| B
+B -->|references| A
+```
+
+In the example below, you define a component called MyMessageSchema under the schemas section to describe the structure of a message. Under the publish operation of myChannel, you reference the MyMessageSchema component using the $ref keyword.
+
+```yaml
+channels:
+  myChannel:
+    publish:
+      message:
+        $ref: '#/components/schemas/MyMessageSchema'
+components:
+  schemas:
+    MyMessageSchema:
+      type: object
+      properties:
+        message:
+          type: string
+```
+
+## Another local document
+
+You can use the ref keyword to reference another local document. Ensure the path to the local file is correct and accessible from your main AsyncAPI document.
+
+The image below defines the process of referencing another local document.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[Main AsyncAPI Document]
+B[message-schema.yaml]
+C[components/schemas]
+D[MyMessageSchema]
+
+A -->|references| B
+B -->|contains| C
+C -->|defines| D
+```
+
+In the example below, you reference the component from the message-schema.yaml file.
+
+```yaml
+##### ./message-schema.yaml
+
+components:
+  messages:
+    UserSignup:
+      name: UserSignup
+      title: User signup
+      summary: Action to sign a user up.
+      description: A longer description
+      contentType: application/json
+      payload: null
+```
+
+```yaml
+##### ./asyncapi.B.yaml
```

Comment:
Why the need for `#####` included in this line? Seems like a mistake.

Reply:
We need to indicate file names, since we are using a couple of files.